Tests/OpenCombineTests/PublisherTests/CountTests.swift

```diff
@@ -0,0 +1,169 @@
+import Combine
+#else
+import OpenCombine
+#endif
+
+@available(macOS 10.15, *)
+final class CountTests: XCTestCase {
+
+    static let allTests = [
+        ("testSendsCorrectCount", testSendsCorrectCount),
+        ("testCountWaitsUntilFinishedToSend", testCountWaitsUntilFinishedToSend),
+        ("testDemand", testDemand),
+        ("testNoDemand", testNoDemand),
+        ("testAddingSubscriberRequestsUnlimitedDemand",
+            testAddingSubscriberRequestsUnlimitedDemand)
+    ]
+
+    func testSendsCorrectCount() {
+        let subscription = CustomSubscription()
+        let publisher = CustomPublisher(subscription: subscription)
+        let countPublisher = publisher.count()
+        let tracking = TrackingSubscriber(
+            receiveSubscription: { $0.request(.max(42)) }
+        )
+
+        XCTAssertEqual(tracking.history, [])
+
+        countPublisher.subscribe(tracking)
+        XCTAssertEqual(tracking.history, [.subscription("Count")])
+
+        let sendAmount = Int.random(in: 1...1000)
+        for _ in 0..<sendAmount {
+            _ = publisher.send(3)
+        }
+        XCTAssertEqual(tracking.history, [.subscription("Count")])
+
+        publisher.send(completion: .finished)
+        XCTAssertEqual(tracking.history, [.subscription("Count"),
+                                          .value(sendAmount),
+                                          .completion(.finished)])
+    }
+
+    func testCountWaitsUntilFinishedToSend() {
+        let subscription = CustomSubscription()
+        let publisher = CustomPublisher(subscription: subscription)
+        let countPublisher = publisher.count()
+        let tracking = TrackingSubscriber(
+            receiveSubscription: { $0.request(.max(42)) }
+        )
+
+        countPublisher.subscribe(tracking)
+
+        _ = publisher.send(1)
+        XCTAssertEqual(tracking.history, [.subscription("Count")])
+
+        _ = publisher.send(2)
+        XCTAssertEqual(tracking.history, [.subscription("Count")])
+
+        _ = publisher.send(0)
+        XCTAssertEqual(tracking.history, [.subscription("Count")])
+
+        publisher.send(completion: .finished)
+        XCTAssertEqual(tracking.history, [.subscription("Count"),
+                                          .value(3),
+                                          .completion(.finished)])
+    }
+
+    func testDemand() {
+        let subscription = CustomSubscription()
+        let publisher = CustomPublisher(subscription: subscription)
+        let countPublisher = publisher.count()
+        var downstreamSubscription: Subscription?
+        let tracking = TrackingSubscriber(
+            receiveSubscription: {
+                $0.request(.max(42))
+                downstreamSubscription = $0
+            },
+            receiveValue: { _ in .max(4) }
+        )
+
+        countPublisher.subscribe(tracking)
+
+        XCTAssertNotNil(downstreamSubscription)
+        dump(type(of: downstreamSubscription!))
```

Comment:
I've accidentally left this `dump` in one of the test classes — it usually helps to shed some light on Combine internals. Let's remove it.

Reply:
Got it.